src/js/components/TextInput/TextInput.js

```diff
@@ -217,23 +217,30 @@ class TextInput extends Component {
     }
   };
 
-  onClickSuggestion = suggestion => {
-    const { forwardRef, onSelect } = this.props;
-    this.setState({ showDrop: false });
+  onClickSuggestion = (suggestion, event) => {
+    const { onSelect } = this.props;
+    this.setState({ showDrop: false, activeSuggestionIndex: -1 });
     if (onSelect) {
-      onSelect({ target: (forwardRef || this.inputRef).current, suggestion });
+      // TODO: needed for backwards compatibility sake
+      /* eslint-disable no-param-reassign */
+      event.suggestion = suggestion;
+      /* eslint-enable no-param-reassign */
+      onSelect(event);
     }
   };
 
   onSuggestionSelect = event => {
-    const { forwardRef, onSelect, suggestions } = this.props;
+    const { onSelect, suggestions } = this.props;
     const { activeSuggestionIndex } = this.state;
-    this.setState({ showDrop: false });
+    this.setState({ showDrop: false, activeSuggestionIndex: -1 });
     if (activeSuggestionIndex >= 0) {
       event.preventDefault(); // prevent submitting forms
-      const suggestion = suggestions[activeSuggestionIndex];
+      // TODO: needed for backwards compatibility sake
+      /* eslint-disable no-param-reassign */
+      event.suggestion = suggestions[activeSuggestionIndex];
+      /* eslint-enable no-param-reassign */
       if (onSelect) {
-        onSelect({ target: (forwardRef || this.inputRef).current, suggestion });
+        onSelect(event);
```

Comment:
I think this will be a breaking change. If the caller is doing:
`onSelect="{({ target }) => this.setState({ x: target.value })}"`
this will no longer work because the event target is now the suggestion button.